lib/src/interface/common.dart

```diff
@@ -108,10 +114,24 @@ String? getExecutablePath(
       return path;
     }
   }
-  if (errorOnNull) {
-    throw ArgumentError('Failed to resolve $command to an executable.\n'
-        'workingDirectory: $workingDirectory, '
-        'candidates: ${candidates.length}');
+  if (throwOnFailure) {
+    if (foundCandidates.isNotEmpty) {
+      throw ProcessPackageExecutableNotFoundException(
+        executable,
+        message:
+            'Found candidates, but failed to resolve $executable to an executable.',
```

Comment:
The tool can post process this, but consider changing the messaging to include permission issues

Reply:
OK.  Something like "Found candidates, but lacked sufficient permissions to execute $executable."?

Reply:
SGTM

Reply:
And hopefully the tool won't need to post process the string: can't it check for a candidate list in the exception and be able to tell?

Reply:
definitely not the string, just the exception list. But for other users, it is a nice improvement